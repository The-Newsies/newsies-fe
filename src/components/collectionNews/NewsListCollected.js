import React from 'react';
import PropTypes from 'prop-types';
import styles from '../news/NewsList.css';
import NewsItemCollected from './NewsItemCollected';

export default function NewsItemsList({ news }) {
  const newsList = news.map(article => (
    <li key={article.url}>
      <NewsItemCollected article={article}/>
    </li>
  ));

  return (
    <ul className={styles.container}>
      {newsList}
    </ul>
  );
}

NewsItemsList.propTypes = {
  news: PropTypes.array.isRequired
};
