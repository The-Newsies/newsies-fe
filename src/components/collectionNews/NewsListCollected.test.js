import React from 'react';
import { shallow } from 'enzyme';
import NewsListCollected from './NewsListCollected';

describe('NewsListCollected component', () => {
  it('renders NewsListCollected', () => {
    const wrapper = shallow(<NewsListCollected 
      news={[{
        title: 'Title',
        description: 'description',
        urlToImage: 'image url',
        publishedAt: 'date',
        source: {
          name: 'name',
          id: 'ddgdgd'
        },
        url: 'sourceLink'
      }]}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});