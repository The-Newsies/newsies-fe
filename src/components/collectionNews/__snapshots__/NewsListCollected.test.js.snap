// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsListCollected component renders NewsListCollected 1`] = `
<ul
  className="container"
>
  <li
    key="sourceLink"
  >
    <NewsItemCollected
      article={
        Object {
          "description": "description",
          "publishedAt": "date",
          "source": Object {
            "id": "ddgdgd",
            "name": "name",
          },
          "title": "Title",
          "url": "sourceLink",
          "urlToImage": "image url",
        }
      }
    />
  </li>
</ul>
`;
